packages/runtime/src/controls/rich-text.ts

```diff
@@ -0,0 +1,132 @@
+import { IndexSignatureHack } from '../prop-controllers/descriptors'
+import type * as Slate from 'slate'
+import { BuilderEditMode } from '../utils/constants'
+import { PropController } from '../prop-controllers/instances'
+import { OnChangeParam, Editor } from 'slate-react'
+import { BoxModel } from '../box-model'
+
+export type RichTextControlData = IndexSignatureHack<Slate.ValueJSON>
+
+export const RichTextControlType = 'makeswift::controls::rich-text'
+
+type RichTextControlConfig = {
+  label?: string
+  defaultValue?: RichTextControlData
+}
+
+export type RichTextControlDefinition<C extends RichTextControlConfig = RichTextControlConfig> = {
+  type: typeof RichTextControlType
+  config: C
+}
+
+export function RichText<C extends RichTextControlConfig>(
+  config: C = {} as C,
+): RichTextControlDefinition<C> {
+  return { type: RichTextControlType, config }
+}
+
+export const RichTextControlMessageType = {
+  CHANGE_BUILDER_EDIT_MODE: 'CHANGE_BUILDER_EDIT_MODE',
+  INITIALIZE_EDITOR: 'INITIALIZE_EDITOR',
+  CHANGE_EDITOR_VALUE: 'CHANGE_EDITOR_VALUE',
+  FOCUS: 'FOCUS',
+  BLUR: 'BLUR',
+  UNDO: 'UNDO',
+  REDO: 'REDO',
+  CHANGE_LAYOUT: 'CHANGE_LAYOUT',
+} as const
+
+type ChangeBuilderEditModeRichTextControlMessage = {
+  type: typeof RichTextControlMessageType.CHANGE_BUILDER_EDIT_MODE
+  editMode: BuilderEditMode
+}
+
+type InitializeEditorRichTextControlMessage = {
+  type: typeof RichTextControlMessageType.INITIALIZE_EDITOR
+  value: Slate.ValueJSON
+}
+
+type ChangeEditorValueRichTextControlMessage = {
+  type: typeof RichTextControlMessageType.CHANGE_EDITOR_VALUE
+  value: Slate.ValueJSON
+}
+
+type FocusRichTextControlMessage = { type: typeof RichTextControlMessageType.FOCUS }
+
+type BlurRichTextControlMessage = { type: typeof RichTextControlMessageType.BLUR }
+
+type UndoRichTextControlMessage = { type: typeof RichTextControlMessageType.UNDO }
+
+type RedoRichTextControlMessage = { type: typeof RichTextControlMessageType.REDO }
+
+type BoxModelChangeRichControlMessage = {
+  type: typeof RichTextControlMessageType.CHANGE_LAYOUT
+  payload: { layout: BoxModel | null }
```

Comment:
For consistency's sake, since we use the term "box model" in all of the runtime instead of "layout", can we change the payload field from `layout` to `boxModel`?